js/base/Exchange.js

```diff
@@ -1304,6 +1301,10 @@ module.exports = class Exchange {
                 trade['fee'] = tradeFee;
             }
         }
+        const tradeFees = this.safeValue (trade, 'fees', []);
+        for (let j = 0; j < tradeFees.length; j++) {
+            trade['fees'][j]['cost'] = this.safeNumber (trade['fees'][j], 'cost');
+        }
```

Comment:
i'm really not trying to be pesky - but i think if you fix it in exchange.js - you'll need to replicate that for Python and php - as exchange.js will not be transpiled.

Reply:
> i'm really not trying to be pesky - but i think if you fix it in exchange.js - you'll need to replicate that for Python and php - as exchange.js will not be transpiled.

@xmatthias not anymore, base js file is now partially transpiled to py/php (actually, mostly transpiled) since recently )

Reply:
oh that's nice :+1:

Reply:
> i'm really not trying to be pesky - but i think if you fix it in exchange.js - you'll need to replicate that for Python and php - as exchange.js will not be transpiled.

Nope, your notes are more then welcome. Just as a good news, we had departed the transpilable codes inside base exchange files, from here and down below as you see: https://github.com/ccxt/ccxt/commit/e98328d46f2b7df6539f22f762dbe744ba317742#diff-306362d145a5405ce27b41bc5613c5371dcbd0857a90cfc7be651bf584db6656R758